scan-command/src/main/java/io/ballerina/scan/internal/ReporterImpl.java

```diff
@@ -58,7 +59,7 @@ class ReporterImpl implements Reporter {
     public void reportIssue(Document reportedDocument, Location location, int ruleId) {
         Rule rule = rules.get(ruleId);
         if (rule == null) {
-            throw new RuleNotFoundException(ruleId);
+            throw new RuntimeException(DiagnosticLog.error(DiagnosticCode.RULE_NOT_FOUND, ruleId));
```

Comment:
Do we need a custom exception for these?

Reply:
What I suggested was having something like ScanToolException for all scan tool related runtime exceptions.